packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -558,6 +572,9 @@ export abstract class FluidDataStoreContext
 		return this.channelP;
 	}
 
+	/**
+	 * ???
```

Comment:
I think this is supposed to find the factory which would be used to construct the DataStore this context is for.

It therefor consults the parent context's registry to find what would be used to construct this, if loading the datastore from a document?

Reply:
Pretty much. When creating a data store (see https://github.com/microsoft/FluidFramework/blob/35ffad29defe6c768aeed78373c5db1fbb6d110c/packages/runtime/container-runtime/src/containerRuntime.ts#L3592), we allow the package path to either be a single string or an array of string. 
If it's a single string, the registry entry for this data store must be in the root (container runtime's) registry.
If it's an array of strings, it allows applicatons to create a virtual hierarchy of where this data store is. Basically, it starts at the root (container runtime), finds a data store for the first entry in the array in the root registry, then finds a data store for the second entry in the array in this data store's registry and so on.

Reply:
When you say "package path" do you mean "this.pkg"?

If so, it can't be a single string, its an array of strings or undefined.

So I don't understand how the single string case is relevant here.

Related: the doc you suggested for this.phk reads like its a single string, but it isn't.

Reply:
Sure, `this.pkg` is a string array and I believe it's that way for simplicity of usage. If a consumer passes a single string as the package when creating a data store, it is converted to an array with single item and then stored here.

If we want consistency across the parameter in the API used to create a data store and how that is stored in `this.pkg`, we could make their types the same. Personally, I don't think it's needed but it should be simple enough to update `this.pkg` as it's internal.